pkg/ipam/crd.go

```diff
@@ -323,6 +323,54 @@ func (n *nodeStore) updateLocalNodeResource(node *ciliumv2.CiliumNode) {
 			}
 		}
 	}
+
+	releaseUpstreamSyncNeeded := false
+	// ACK or NACK IPs marked for release by the operator
+	for ip, status := range n.ownNode.Status.IPAM.ReleaseIPs {
+		if status != ipamOption.IPAMMarkForRelease || n.ownNode.Spec.IPAM.Pool == nil {
+			continue
+		}
+		// NACK the IP, if this node doesn't own the IP
+		if _, ok := n.ownNode.Spec.IPAM.Pool[ip]; !ok {
+			n.ownNode.Status.IPAM.ReleaseIPs[ip] = ipamOption.IPAMDoNotRelease
+			continue
+		}
+		// Retrieve the appropriate allocator
+		var allocator *crdAllocator
+		var ipFamily Family
+		if ipAddr := net.ParseIP(ip); ipAddr != nil {
+			if ipAddr.To4() != nil {
+				ipFamily = IPv4
+			} else {
+				ipFamily = IPv6
+			}
```

Comment:
tiny nit (feel free to ignore): 
```suggestion
			ipFamily = DeriveFamiliy(ipAddr)
```

Reply:
Updated this.